Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -21,48 +21,70 @@ namespace ROS2::Utils
     {
         AZStd::vector<AZStd::string> modifiedLinks;
         using namespace AZ::rapidxml;
-        AZStd::vector<xml_node<>*> links;
+
         for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
         {
-            if (!link->first_node("inertial"))
+            bool modified = false;
+
+            auto name_xml = link->first_attribute("name");
+            AZStd::string name = "unknown_link";
+            if (name_xml)
             {
-                links.push_back(link);
+                name = name_xml->value();
             }
-        }
 
-        for (auto* link : links)
-        {
-            xml_node<>* inertial = urdf->document()->allocate_node(node_element, "inertial");
+            auto inertial = link->first_node("inertial");
+            if (!inertial)
+            {
+                AZ_Warning("URDF", false, "Missing inertial tag in link %s, applying default tag.", name.c_str());
+                inertial = urdf->document()->allocate_node(node_element, "inertial");
+                link->append_node(inertial);
+                modified = true;
+            }
 
-            xml_node<>* mass = urdf->document()->allocate_node(node_element, "mass");
-            mass->append_attribute(urdf->document()->allocate_attribute("value", "1."));
-            inertial->append_node(mass);
-            xml_node<>* inertia = urdf->document()->allocate_node(node_element, "inertia");
+            auto mass = inertial->first_node("mass");
+            if (!mass)
+            {
+                mass = urdf->document()->allocate_node(node_element, "mass");
```

Comment:
A similar warning would be helpful here as well, but only if `modified` is `false`, since i am assuming a default tag won't have this.
```suggestion
                if (!modified)
                {
                    AZ_Warning("URDF", false, "Missing mass tag inside the inertial tag link %s, applying default mass tag.", name.c_str());
                }
                mass = urdf->document()->allocate_node(node_element, "mass");
                
```

Reply:
When the 'inertial' tag is missing, also all subtags are added.
But since they are always added in this case, I think it is unnecessary to warn user about each one separately. 

I added a warning in case the `inertial` tag is present, but the subtag is missing.